simpeg_drivers/plate_simulation/sweep/driver.py

```diff
@@ -119,24 +161,74 @@ def run_worker(uid: str, data: dict, h5file: Path, workdir: Path | None):
             return
 
         shutil.copy(h5file, workerfile)
-        with Workspace(workerfile, mode="r+") as worker:
+        with Workspace(workerfile, mode="r+") as workspace:
             plate_simulation = next(
                 group
-                for group in worker.groups
+                for group in workspace.groups
                 if isinstance(group, SimPEGGroup | UIJsonGroup)
                 and "plate_simulation.driver" in group.options.get("run_command")
             )
 
-            ifile = InputFile(ui_json=plate_simulation.options, validate=False)
-            for key, value in data.items():
-                ifile.set_data_value(key, value)
-            options = PlateSimulationOptions.build(
-                ifile.data, geoh5=worker, out_group=plate_simulation
-            )
-            options.write_ui_json(workdir / f"{uid}.ui.json")
-            PlateSimulationDriver.start(workdir / f"{uid}.ui.json")
+            opt_dict = workspace.promote(flatten(plate_simulation.options))
+            opt_dict["geoh5"] = workspace
+            opt_dict["out_group"] = None
+            opt_dict.update(data)
+            options = PlateSimulationOptions.build(opt_dict)
+            plate_sim = PlateSimulationDriver(options, workers=[worker])
+            plate_sim.run()
+
+        del plate_sim
+        return None
+
+
+def trial_runs(
+    trials: list[dict],
+    h5file: Path,
+    workdir: Path | None,
+    worker: tuple[str] | None = None,
+):
+    """
+    Loop through a list of trials and run a worker for each unique parameter set.
+    """
+    for kwargs in trials:
+        PlateSweepDriver.run_trial(kwargs, h5file, workdir, worker=worker)
 
 
 if __name__ == "__main__":
-    file = Path(sys.argv[1]).resolve()
-    PlateSweepDriver.start(file)
+    file = Path(sys.argv[1])
```

Comment:
Do we need to repeat this here?  Can the BaseDriver definition take care of this?